python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -180,6 +181,65 @@ def test_get_conda_env_dir(tmp_path):
         assert (env_dir == str(tmp_path / "envs" / "tf2"))
 
 
+def test_conda_create_task(shutdown_only):
```

Comment:
These tests will fail if anyone tries to make a PR that is incompatible with the latest nightly (for example, a PR that changes the number of arguments in `worker.connect()`.), because the workers will be using the Ray nightly but the driver and Raylet will be using the code from the PR.  I couldn't think of a workaround, open to ideas here

Reply:
Replacing the nightly with a local install (`-e /ray/python`) is too slow I think, because it has to build all of Ray.  It also depends on the directory where the Ray source is located, which differs for the developer and for CI

Reply:
Seems like not a huge deal for now because the tests are very simple.

@simon-mo thoughts here? Can we use the same version that's installed in CI easily?

Reply:
can you install ray from file in the conda env? essentially doing a `conda activate env; pip install ray/python`. The install should be fast because all C++ stuff is cached.

Reply:
relying on nightly will lead to very flaky tests on PR. Ray Core PR can easily change both Python and C++ code at the same time.